eslint-bridge/src/rules/decorators/no-empty-function-decorator.ts

```diff
@@ -19,24 +19,83 @@
  */
 // https://sonarsource.github.io/rspec/#/rspec/S1186/javascript
 
+import estree from 'estree';
 import { Rule } from 'eslint';
-import * as estree from 'estree';
-import { interceptReport } from '../../utils';
+import { FunctionNodeType, interceptReport, isFunctionNode, isIdentifier } from '../../utils';
 import { suggestEmptyBlockQuickFix } from './no-empty-decorator';
 
-type FunctionLike =
-  | estree.ArrowFunctionExpression
-  | estree.FunctionDeclaration
-  | estree.FunctionExpression;
+type RuleFunctionNode = FunctionNodeType & Rule.Node;
+
+function isRuleFunctionNode(node: estree.Node): node is RuleFunctionNode {
+  return isFunctionNode(node) && 'parent' in node;
+}
 
 // core implementation of this rule does not provide quick fixes
 export function decorateNoEmptyFunction(rule: Rule.RuleModule): Rule.RuleModule {
   rule.meta!.hasSuggestions = true;
-  return interceptReport(rule, (context, reportDescriptor) => {
-    const func = (reportDescriptor as any).node as FunctionLike;
-    const name = reportDescriptor.data!.name;
-    const openingBrace = context.getSourceCode().getFirstToken(func.body)!;
-    const closingBrace = context.getSourceCode().getLastToken(func.body)!;
-    suggestEmptyBlockQuickFix(context, reportDescriptor, name, openingBrace, closingBrace);
-  });
+  return interceptReport(rule, reportWithQuickFixIfApplicable);
+}
+
+export function reportWithQuickFixIfApplicable(
+  context: Rule.RuleContext,
+  reportDescriptor: Rule.ReportDescriptor,
+) {
+  if (!('node' in reportDescriptor) || !isRuleFunctionNode(reportDescriptor.node)) {
+    return;
+  }
+
+  const functionNode = reportDescriptor.node;
+  if (isApplicable(functionNode)) {
+    reportWithQuickFix(context, reportDescriptor, functionNode);
+  }
+}
+
+// This function limits the issues to variable/function/method declarations which name is not like /^on[A-Z].
+// Any lambda expression or arrow function is thus ignored.
+function isApplicable(functionNode: RuleFunctionNode): boolean {
+  // Matches identifiers like onClick and more generally onXxx
+  function isCallbackIdentifier(node: estree.Node | null): boolean {
+    return node !== null && isIdentifier(node) && /^on[A-Z]/.test(node.name);
+  }
+
+  // Matches: function foo() {}
+  // But not: function onClose() {}
+  function isFunctionDeclaration(): boolean {
+    return functionNode.type === 'FunctionDeclaration' && !isCallbackIdentifier(functionNode.id);
+  }
+
+  // Matches: Class A { foo() {} }
+  // But not: Class A { onClose() {} }
```

Comment:
`class` should be lowercase
```suggestion
  // Matches: class A { foo() {} }
  // But not: class A { onClose() {} }
```